cake/versioning.cake

```diff
@@ -0,0 +1,141 @@
+using System.Text.RegularExpressions;
+
+public class BuildVersion
+{
+    private ISetupContext _context;
+    private GitVersion _gitVersion;
+
+    // NOTE: This is complicated because (1) the user may have specified 
+    // the package version on the command-line and (2) GitVersion may
+    // or may not be available. We'll work on solving (2) by getting
+    // GitVersion to run for us on Linux, but (1) will alwas remain.
```

Comment:
```suggestion
    // GitVersion to run for us on Linux, but (1) will always remain.
```

Reply:
Actually, I should take most of this out. I got it running on Linux!